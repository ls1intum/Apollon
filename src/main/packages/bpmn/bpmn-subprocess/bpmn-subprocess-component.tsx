import React, { FunctionComponent } from 'react';
import { BPMNSubprocess } from './bpmn-subprocess';
import { ThemedPolyline, ThemedRect } from '../../../components/theme/themedComponents';
import { Multiline } from '../../../utils/svg/multiline';

export const BPMNSubprocessComponent: FunctionComponent<Props> = ({ element, fillColor }) => (
  <g>
    <ThemedRect
      rx={10}
      ry={10}
      width="100%"
      height="100%"
      strokeColor={element.strokeColor}
      fillColor={fillColor || element.fillColor}
    />
    <ThemedRect
      x={element.bounds.width / 2 - 7}
      y={element.bounds.height - 14}
      width={14}
      height={14}
      strokeColor={element.strokeColor}
    />
    <ThemedPolyline
      points={`${element.bounds.width / 2 - 4} ${element.bounds.height - 7}, ${element.bounds.width / 2 + 4} ${
        element.bounds.height - 7
      }`}
    />
    <ThemedPolyline
      points={`${element.bounds.width / 2} ${element.bounds.height - 11}, ${element.bounds.width / 2} ${
        element.bounds.height - 3
      }`}
    />
    <Multiline
      x={element.bounds.width / 2}
      y={element.bounds.height / 2}
      width={element.bounds.width}
      height={element.bounds.height}
      fontWeight="bold"
      fill={element.textColor}
      lineHeight={16}
      capHeight={11}
    >
      {element.name}
    </Multiline>
  </g>
);

interface Props {
  element: BPMNSubprocess;
  fillColor?: string;
}
