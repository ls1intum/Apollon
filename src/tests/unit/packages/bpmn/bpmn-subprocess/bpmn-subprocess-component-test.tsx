import { wrappedRender } from '../../../test-utils/render';
import * as React from 'react';
import { BPMNSubprocess } from '../../../../../main/packages/bpmn/bpmn-subprocess/bpmn-subprocess';
import { BPMNSubprocessComponent } from '../../../../../main/packages/bpmn/bpmn-subprocess/bpmn-subprocess-component';

it('render the bpmn-subprocess-component', () => {
  const subprocess: BPMNSubprocess = new BPMNSubprocess({ name: 'SyntaxTreeTerminal' });
  const { getByText, baseElement } = wrappedRender(
    <svg>
      <BPMNSubprocessComponent element={subprocess} />
    </svg>,
  );
  expect(getByText(subprocess.name)).toBeInTheDocument();
  expect(baseElement).toMatchSnapshot();
});
