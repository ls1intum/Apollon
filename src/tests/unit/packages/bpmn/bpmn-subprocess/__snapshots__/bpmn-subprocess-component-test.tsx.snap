// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the bpmn-subprocess-component 1`] = `
<body>
  <div>
    <svg>
      <g>
        <rect
          class="sc-fqkvVR kprGcp"
          fill="white"
          height="100%"
          rx="10"
          ry="10"
          stroke="black"
          width="100%"
        />
        <rect
          class="sc-fqkvVR zazVk"
          fill="transparent"
          height="14"
          stroke="black"
          width="14"
          x="73"
          y="86"
        />
        <polyline
          class="sc-aXZVg bUvEJV"
          fill="white"
          points="76 93, 84 93"
          stroke="black"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
        <polyline
          class="sc-aXZVg bUvEJV"
          fill="white"
          points="80 89, 80 97"
          stroke="black"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
        <text
          font-weight="bold"
          height="100"
          pointer-events="none"
          text-anchor="middle"
          width="160"
          x="80"
          y="50"
        >
          <tspan
            dy="5.5"
            x="80"
          >
            SyntaxTreeTerminal
          </tspan>
        </text>
      </g>
    </svg>
  </div>
</body>
`;
